klog.go

```diff
@@ -831,6 +831,7 @@ func SetLogger(logr logr.Logger) {
 func SetOutput(w io.Writer) {
 	logging.mu.Lock()
 	defer logging.mu.Unlock()
+	logging.toStderr = false
```

Comment:
feels like a change in behavior. consumer using the current default (logtostderr = true) would not get it there anymore.
is it necessary for the examples of v1 + v2?

Reply:
Yes, it's needed @neolit123

Reply:
By default [toStdErrr is true](https://github.com/kubernetes/klog/blob/master/klog.go#L411). Since we will not call InitFlags, we end up skipping whatever we set up using SetOutput/SetOutputBySeverity because of this [piece of code](https://github.com/kubernetes/klog/blob/master/klog.go#L873-L875)

Reply:
this is tricky...at minimum it needs a release note and even after that i'm not sure it should be done like so. are there workarounds without changing the user facing behavior, e.g. a new private bool var somewhere?

Reply:
+1 to release note. if folks are NOT using those methods, they won't see any changes.

Reply:
ok, understood. i'm going to defer my LGTM to an approver of klog.
e.g. @liggitt commented on the same problem earlier [here](https://github.com/kubernetes/klog/pull/142#discussion_r413056281).

Reply:
https://github.com/kubernetes/klog/pull/100 proposed control over this variable...

I'm quite concerned that the existing methods have many unexpected side effects already, and people have worked around them in various ways, and that breaking those workarounds will cause significant pain downstream.

Reply:
The current mix of flag-controlled aspects and programmatically controlled aspects is quite difficult to navigate. I don't think adding more side effects to the programmatic methods is what we want to do. Thinking as a consumer, the more 1:1 a setter is, the better.

In klog v1:

Log to stderr in all of the following cases:
* toStderr is true (only settable via flag, defaults to true)
* alsoToStderr is true (only settable via flag, defaults to false)
* go's standard log is being written to via something installed with CopyStandardLogTo(...)
* log message threshold is >= stderrthreshold (only settable via flag, defaults to error)

Log to file output **IFF** toStderr is false (only settable via flag, defaults to true) and at least one of the following is true:
  * `log_file` is set (only settable via flag, defaults to "")
  * SetOutput() was called
  * SetOutputBySeverity() was called and matches the message's severity

klog/v2 makes this more complicated by adding the logr interface as a third case, which takes precedence over all flags and outputs, as far as I can tell.

Reply:
gotcha, let's merge https://github.com/kubernetes/klog/pull/100 and i'll remove the hacks here

Reply:
@liggitt DONE! merged #100 and updated the examples to use that new method to disable toStdErr